### PR TITLE
[R20-1508] Added new tide_data_pipeline module.

### PR DESCRIPTION
**Pull request by MdNadimHossain**
### JIRA
https://digital-vic.atlassian.net/browse/R20-1508

### Changes
1. Supporting data pipeline from core
2. separating in a submodule
3. Adding dependency module in composer

### Test branch 
https://github.com/dpc-sdp/content-reference-sdp-vic-gov-au/pull/326
After build the link will be - https://nginx-php.pr-326.content-reference-sdp-vic-gov-au.sdp4.sdp.vic.gov.au/

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -4,6 +4,11 @@
     "patches": {
       "drupal/search_api": {
         "Provide drush support to index a specific entity - https://www.drupal.org/project/search_api/issues/3016809": "https://www.drupal.org/files/issues/2020-02-08/search_api-index-specific-items-3016809-09.patch"
+      },
+      "drupal/key": {
+        "KeyProviderInterface::getKeyValue() doesn't always return a string - https://www.drupal.org/project/key/issues/3356052#comment-15030602": "https://git.drupalcode.org/project/key/-/merge_requests/11/diffs.patch",
+        "Deprecated Function: strlen(): Passing null to parameter #1 ($string) - https://www.drupal.org/project/key/issues/3323238": "https://www.drupal.org/files/issues/2023-02-01/3323238-6.patch",
+        "Error: Call to a member function getKeyValue on null - https://www.drupal.org/project/key/issues/3385523": "https://www.drupal.org/files/issues/2023-09-06/3385523-3.patch"
```
Comment (MdNadimHossain) [inline composer.json:11]: Created this patch new. Please review.

Comment (krakerag) [reply composer.json:11]: Added one comment on the patch over there, nothing major just being picky :P

File: composer.json [hunk 2]
```diff
@@ -24,7 +29,10 @@
     "dpc-sdp/tide_core": "^3.0.0",
     "dpc-sdp/tide_landing_page": "^3.0.0",
     "drupal/elasticsearch_connector": "^7.0",
-    "drupal/search_api": "^1.11"
+    "drupal/search_api": "^1.11",
+    "drupal/data_pipelines": "^1.0@alpha",
+    "drupal/data_pipelines_elasticsearch": "^1.0@alpha",
```
Comment (krakerag) [inline composer.json:34]: Note there may be an updated version coming from my fix for D10 bulk endpoint, or we may have to add another patch if it's not released in time.

**Review by krakerag — APPROVED** [review]
Looks good.

One thing to note - will we have to plan a migration task for Cameras Save Lives? Or is this going to be in content vic in addition to CSL? It uses different machine names so I think it will be just a second configured pipeline available and CSL won't even care about this being here.

File: composer.json (hunk 1 above)
Comment (MdNadimHossain) [reply composer.json]: Yup see the point, thanks fro that. I have Updated the patch :)

File: composer.json (hunk 2 above)
Comment (MdNadimHossain) [reply composer.json:34]: Cool, I will keep an eye while updating to drupal 10, thanks for the heads up

**Comment by MdNadimHossain** [comment]
> 

@krakerag Shouldn't require any migration, cause the content-vic camera saves live should stay as a different data set using the camera save lives data pipeline. The new one that got added in this PR will be available as a option when creating a new data set or even updating the exisiting. It should appear in the drop down of a data set in the datapipeline selection field.

**Pull request merged**